crates/core_arch/src/x86/avx512f.rs

```diff
@@ -87,6 +98,302 @@ pub unsafe fn _mm512_setr_epi32(
     transmute(r)
 }
 
+/// Gather double-precision (64-bit) floating-point elements from memory using 32-bit indices.
+///
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm512_i32gather_pd)
+#[inline]
+#[target_feature(enable = "avx512f")]
+#[cfg_attr(test, assert_instr(vgatherdpd, scale = 1))]
+pub unsafe fn _mm512_i32gather_pd(offsets: __m256i, slice: *const u8, scale: i32) -> __m512d {
+    let zero = _mm512_setzero_pd().as_f64x8();
+    let neg_one = -1;
+    let slice = slice as *const i8;
+    let offsets = offsets.as_i32x8();
+    macro_rules! call {
+        ($imm8:expr) => {
+            vgatherdpd(zero, slice, offsets, neg_one, $imm8)
+        };
+    }
+    let r = constify_imm8!(scale, call);
```

Comment:
According to the intrinsic documentation, only 1, 2, 4, 8 are valid values for the scale. You should use a custom `constify` macro that handles this.

Also you should use `#[rustc_args_required_const(<arg index of scale>)]` since `scale` is required to be a compile-time constant.

Reply:
Done.

The new macro panics if it gets a value outside of 1, 2, 4, 8. Arguably, it should catch errors at compile time, but the one thing I tried `std::compile_error` would not compile at all. If you have any suggestions, I'm more than happy to fix this.

I've added the new macro to `src/x86/macros.rs` because we should probably change the other existing gather intrinsics to use the new macro as well. That would be a backwards incompatible change, but it would only affect broken code. I don't know what the official policy is in cases like this. I'm happy to make the change in a separate PR if you would like.